src/storage/DataAccessorBasedStore.ts

```diff
@@ -347,8 +348,22 @@ export class DataAccessorBasedStore implements ResourceStore {
    * @param slug - Slug to use for the new URI.
    */
   protected createURI(container: ResourceIdentifier, isContainer: boolean, slug?: string): ResourceIdentifier {
-    return { path:
-        `${ensureTrailingSlash(container.path)}${slug ? trimTrailingSlashes(slug) : uuid()}${isContainer ? '/' : ''}` };
+    let path = ensureTrailingSlash(container.path);
+    path += slug ? this.cleanSlug(slug) : uuid();
+    path += isContainer ? '/' : '';
+    return { path };
+  }
+
+  /**
+   * Clean http Slug to be compatible with the server. Makes sure there are no unwanted characters
+   * e.g.: cleanslug('&/%26') returns '%26%2F%26'
+   * @param slug - the slug to clean
+   */
+  protected cleanSlug(slug: string): string {
+    if (trimTrailingSlashes(slug).includes('/')) {
+      throw new BadRequestHttpError('Slugs should not contain slashes');
+    }
+    return toCanonicalUriPath(trimTrailingSlashes(slug));
```

Comment:
Avoid trimming twice. Suggestion:

```suggestion
    if (/\/[^\/]/.test(slug)) {
      throw new BadRequestHttpError('Slugs should not contain slashes');
    }
    return toCanonicalUriPath(trimTrailingSlashes(slug));
```

Reply:
I would just error even if the slashes are trailing in the slug tbh. I don't think we have a relevant use case for allowing that one specifically.